californium-core/src/test/java/org/eclipse/californium/core/test/OptionTest.java

```diff
@@ -211,25 +333,26 @@ public void testLocationOptions() {
 	
 	@Test
 	public void testArbitraryOptions() {
+	    // arbitrary options not present in OptionNumberRegistry are not allowed anymore.
 		OptionSet options = new OptionSet();
 		options.addETag(new byte[] {1, 2, 3});
 		options.addLocationPath("abc");
-		options.addOption(new Option(7));
-		options.addOption(new Option(43));
-		options.addOption(new Option(33));
-		options.addOption(new Option(17));
+		options.addOption(new Option(OptionNumberRegistry.parse(7)));
+		//options.addOption(new Option(OptionNumberRegistry.parse(43)));
+		//options.addOption(new Option(OptionNumberRegistry.parse(33)));
+		options.addOption(new Option(OptionNumberRegistry.parse(17)));
 
 		// Check that options are in the set
 		Assert.assertTrue(options.hasOption(OptionNumberRegistry.ETAG));
```

Comment:
assertions commented in this test are all invalid now, as they all refer to unexistent OptionNumberRegistry values.  If arbitrary option is to be used, either define them in OptionNumberRegistry (easier now with the refactoring) or a new patch must be added.  I think this is the best approach to the problem.